server/src/main/java/keywhiz/service/daos/GroupDAO.java

```diff
@@ -93,22 +118,57 @@ public void deleteGroup(Group group) {
           .set(SECRETS.OWNER, (Long) null)
           .where(SECRETS.OWNER.eq(group.getId()))
           .execute();
+      DSL.using(configuration)
+          .update(GROUPS)
+          .set(GROUPS.OWNER, (Long) null)
+          .where(GROUPS.OWNER.eq(group.getId()))
+          .execute();
     });
```

Comment:
In the absence of enforced foreign keys, we're forced to manually clean up dangling references to a deleted group.